File: mobile/android/LiveRecallGlasses/app/src/main/kotlin/com/liverecall/glasses/ui/components/ConnectionPill.kt
package com.liverecall.glasses.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.liverecall.glasses.livekit.SessionController
import com.liverecall.glasses.ui.theme.Tokens

/**
 * Status pill matching the .pill / .pill.live styling from
 * phone/glasses.html. Color follows the connection state.
 */
@Composable
fun ConnectionPill(status: SessionController.Status) {
    val (label, bg, fg) = when (status) {
        is SessionController.Status.Idle ->
            Triple("DISCONNECTED", Tokens.PanelBorder, Tokens.Muted)
        is SessionController.Status.Connecting ->
            Triple("CONNECTING…", Tokens.Warn.copy(alpha = 0.15f), Tokens.Warn)
        is SessionController.Status.Live ->
            Triple("LIVE · POV", Tokens.Accent.copy(alpha = 0.18f), Tokens.Accent)
        is SessionController.Status.Error ->
            Triple("ERROR", Tokens.Err.copy(alpha = 0.18f), Tokens.Err)
    }
    Text(
        text = label,
        color = fg,
        fontSize = 11.sp,
        fontWeight = FontWeight.SemiBold,
        modifier = Modifier
            .clip(CircleShape)
            .background(bg)
            .padding(horizontal = 10.dp, vertical = 4.dp)
    )
}

File: mobile/ios/LiveRecallGlasses/LiveRecallGlasses/Views/ConnectionPill.swift
// ConnectionPill.swift
//
// Status pill in the top-right of the screen. Mirrors the .pill / .pill.live
// / .pill.warn / .pill.err treatment from phone/glasses.html.

import SwiftUI

struct ConnectionPill: View {
    let status: SessionController.Status

    var body: some View {
        Text(label)
            .font(.system(size: 11, weight: .semibold))
            .tracking(0.6)
            .textCase(.uppercase)
            .padding(.horizontal, 10)
            .padding(.vertical, 4)
            .background(background)
            .foregroundColor(foreground)
            .clipShape(Capsule())
    }

    private var label: String {
        switch status {
        case .idle: return "disconnected"
        case .connecting: return "connecting…"
        case .live: return "live · POV"
        case .error: return "error"
        }
    }

    private var background: some View {
        Group {
            switch status {
            case .live:
                Theme.accent.opacity(0.18)
            case .connecting:
                Theme.warn.opacity(0.15)
            case .error:
                Theme.err.opacity(0.18)
            case .idle:
                Color(hex: 0x1F2937)
            }
        }
    }

    private var foreground: Color {
        switch status {
        case .live: return Theme.accent
        case .connecting: return Theme.warn
        case .error: return Theme.err
        case .idle: return Theme.muted
        }
    }
}
